app/s3.py

```diff
@@ -0,0 +1,51 @@
+"""Utilities for working with S3.
+
+In LSST the Docs, ltd-mason is responsible for uploading documentation
+resources to S3. ltd-keeper only really needs to delete resources.
+"""
+
+import logging
+from pprint import pformat
+import boto3
+
+from .exceptions import S3Error
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def delete_directory(bucket_name, root_path):
+    """Delete all objects in the S3 bucket named `bucket_name` that are
+    found in the `root_path` directory.
+
+    Parameters
+    ----------
+    bucket_name : str
+        Name of an S3 bucket.
+    root_path : str
+        Directory in the S3 bucket that will be deleted.
```

Comment:
Should [or shouldn't] `root_path` start with a `/`?  Which ever it is, I'm fairly sure as an end-user I would get it wrong.